### ISSUE TITLE
"object:modified" event doesn't get fired when changing object props using set method

### ISSUE THREAD

**Issue by houssameddine-h** [fabricjs/fabric.js#9418]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.0-beta13

### In What environments are you experiencing the problem?

Firefox, Chrome

### Node Version (if applicable)

None

### Link To Reproduction

https://jsfiddle.net/wg0sar21/1/

### Steps To Reproduce

1. Add a listener to the fabric canvas for the event `object:modified`;
2. Create a new Rect object and add to the canvas;
3. Update the Rect properties using the `set(...)` method.

### Expected Behavior

The `object:modified` event should fire.

### Actual Behavior

The `object:modified` event doesn't get fired.

You mentioned here #8299 that:

> `stateful` was used to fire an object modified every render cycle in which a change in state properties was determiend. All the events that can modify an object already fire an `object:modified` event, and the user has no ability to change colors or state outside developer written code, so there is no really need for event firing here.

But when changing object properties using  `object.set(property, value)`, the `object:modified` event doesn't get fired.

Am I missing something here?

**Comment by asturur** [comment]
consider events as something that have to warn you when the user is changing things using code that is not written by you. For example when modifying an object with controls. The end user has no way to use the set method, so fabric does not need to inform you.

**Comment by houssameddine-h** [comment]
Yeah, but most of the time the user will be changing things using code written by me (using keyboard arrows or text inputs for example), and in all those cases I have to call `Canvas.fire(...)` (or use ShaMan123's proposed solution).
Actually, I'm OK with it. I just thought it was an issue since it was there in older versions.
